cfg/config_test.go

```diff
@@ -1,71 +1,56 @@
 package cfg
 
 import (
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"testing"
 )
 
-func TestLoadConfig(t *testing.T) {
-	_, err := LoadConfig("foobar")
-	if err == nil {
-		t.Error("Loading an invalid config file should return an error")
-	}
-
-	conf, err := LoadConfig(filepath.Join("testdata", "beehive.conf"))
+func TestNew(t *testing.T) {
```

Comment:
We should probably also try initializing with an invalid URL, to verify it returns an error.

Reply:
This could be trickier, because `url.Parse` accepts pretty much any random string, and to preserve backwards compat I opted for accepting things like `/path/to/my/beehive.conf`, `c:\Program Folder\foobar.conf`, etc.

Do you think we should be stricter here and only accept URLs with a valid scheme, host or path, port, etc?

Reply:
We should at least clarify that adding some docs and maybe change the parameter name to something else to make the intention clear.

Reply:
Actually I'm not sure native windows filesystem paths are really accepted by Beehive currently, i.e. `C:\foobar\stuff`. If you run Beehive from master with something like `beehive --config 'c:\foobar\stuff', it'll save the config to `$CWD/C:\foobar\stuff` in a Unix system.

Should we leave better config path handling for a new, dedicated PR where we take into consideration the issues with different OSes?

Reply:
Ended up adding some extra docs, better validation and some extra tests.

Let me know if that works.